src/events/guildAuditLogEntryCreate.ts

```diff
@@ -0,0 +1,224 @@
+				if (timeoutChange.new && timeoutChange.new && typeof timeoutChange.new === 'string') {
+					const timeout = new Date(timeoutChange.new);
+					await handleMemberTimedOut(guild, user, executor, auditLogEntry.reason, timeout);
+				}
+			}
+
+			// * This checks that a user has changed their name
+			const nickChange = auditLogEntry.changes.find(change => change.key === 'nick');
+			// * The types here should always be correct, but I have no idea how to convince the discord.js types of that...
+			if (nickChange && typeof nickChange.new === 'string' && (nickChange.old === undefined || typeof nickChange.old === 'string')) {
+				await handleMemberNicknameChange(guild, user, nickChange.old, nickChange.new);
+			}
+
+			break;
+		}
+		case AuditLogEvent.MemberKick: {
+			await handleMemberKick(auditLogEntry, guild);
+			break;
+		}
+		case AuditLogEvent.MemberBanAdd: {
+			await handleMemberBanAdd(auditLogEntry, guild);
+			break;
+		}
+	}
+}
+
+async function handleMemberTimedOut(guild: Guild, user: User, executor: User, reason: string | null, timeout: Date): Promise<void> {
+	const embed = new EmbedBuilder();
+
+	embed.setColor(0xC0C0C0);
+	embed.setDescription('――――――――――――――――――――――――――――――――――');
+	embed.setFooter({
+		text: 'Pretendo Network',
+		iconURL: guild.iconURL() as string
+	});
+
+	embed.setTitle('Event Type: _Member Timedout_');
+
+	embed.setFields(
+		{
+			name: 'User',
+			value: `<@${user.id}>`
+		},
+		{
+			name: 'User ID',
+			value: user.id
+		},
+		{
+			name: 'Executor',
+			value: `<@${executor.id}>`
+		},
+		{
+			name: 'Executor User ID',
+			value: executor.id
+		},
+		{
+			name: 'Reason',
+			value: reason ?? 'No reason given'
+		},
+		{
+			name: 'Disabled Until',
+			value: `<t:${Math.trunc(timeout.getTime() / 1000)}>`
+		}
+	);
+
+	await sendEventLogMessage(guild, null, embed);
+}
+
+async function handleMemberNicknameChange(guild: Guild, user: User, oldName?: string, newName?: string): Promise<void> {
+	const embed = new EmbedBuilder();
+
+	embed.setColor(0xC0C0C0);
+	embed.setDescription('――――――――――――――――――――――――――――――――――');
+	embed.setFooter({
+		text: 'Pretendo Network',
+		iconURL: guild.iconURL() as string
+	});
+
+	embed.setTitle('Event Type: _Member Nickname Update_');
+
+	embed.setFields(
+		{
+			name: 'User',
+			value: `<@${user.id}>`
+		},
+		{
+			name: 'User ID',
+			value: user.id
+		},
+		{
+			name: 'Old Nickname',
+			value: oldName ?? '(No Nickname)',
+			inline: true
+		},
+		{
+			name: 'New Nickname',
+			value: newName ?? '(No Nickname)',
+			inline: true
+		}
+	);
+
+	await sendEventLogMessage(guild, null, embed);
+}
+
+async function handleMemberKick(auditLogEntry: GuildAuditLogsEntry, guild: Guild): Promise<void> {
+	// * Unsure why just checking the action type doesn't narrow the type here, probably too much type inference going on?
+	const user = (auditLogEntry as unknown as GuildAuditLogsEntry<AuditLogEvent.MemberKick>).target;
+	if (!user) {
+		return;
+	}
```

Comment:
Explicitly typing the entry does narrow the type:

![Screenshot from 2024-07-19 08-33-37](https://github.com/user-attachments/assets/ca928ef2-d30c-4aec-8105-dfe5843c458d)

Reply:
Solved with type guard from above

Reply:
I'm not sure what you mean? This doesn't look changed, and by explicitly typing `auditLogEntry` here you can remove the `as unknown as GuildAuditLogsEntry<AuditLogEvent.MemberKick>`, since it will already be narrowed, like I showed in my screenshot?

Reply:
I don't know if this is just GitHub being weird? I've removed all of the casts from every `user` instance in the file, and I didn't even rebase/force push

Reply:
It must be an issue with the mobile app, on desktop I do see the changes. Odd